### ISSUE TITLE
Behringer UMC2 Sound Quality problem

### ISSUE THREAD

**Issue by parsivan** [Andersama/obs-asio#113]
OBS version: 
27.2.3
Plugin Version:
3.1.0
Soundcard:
Behringer UMC2 & Behringer UMC1820

I want to stream/record a podcast with UMC1820 and I was using the UMC2 because i had that avaliable at the time; to do testing on the setup so there's no problem when we want to stream or record. I have tested my soundcard and my mic (MXL990) and tested versions of ASIO drivers 2.15(stable) and 2.14(stable). i have checked if it was a filter problem and the audio is still bad with a clean signal.

Describing the Bug: (I'm not sure if it's really a bug)
 - The Audio coming from the soundcard sound like a crappy built-in microphone of a budget laptop; muffled sound, very broken.

OBS log:
https://obsproject.com/logs/wV-U5Lq1_qA0CkH

Sample Recordings:(through obs and with DAW)
https://user-images.githubusercontent.com/21005543/160860286-c6325e6d-8a7e-4d84-88b8-79879c591554.mp4

https://user-images.githubusercontent.com/21005543/160860872-10be37e4-527e-4160-b2f8-83b94d9808dc.mp4

**Comment by parsivan** [comment]
It's supposed to be UM2 instead of UMC 2. This was used for my tests and The actual recording is gonna be with UMC1820.

**Comment by Andersama** [comment]
I'd double check you're recording with the device you want, typically people complain about crackling / popping / static, muffled audio is definitely unusual. Since it sounds like you're saying you've got two cards to record from, that's the most obvious thing. If at all possible try recording w/ a daw and obs at the same time. You can try the virtual asio routing workarounds if needed.

**Comment by parsivan** [comment]
yeah in the past I recorded the audio and video separately(obs + Ableton) but we wanted to make things easier by streaming it directly. I still have not tested the other sound card yet but I'm going to see if it works. 
will this virtual asio routing fix this problem? can you tell me more? I'm new to audio routing and stuff.

**Comment by Andersama** [comment]
You can read about some of this in the wiki section, but a quick rundown is that not every device / asio driver is completely stable. Some device drivers for example have problems when multiple applications try using the asio device at the same time. Some have odd bugs where certain functions need to be called in a particular order, we've even had one whose 64 bit version was stable where the 32 bit one was not.

To circumvent issues with multiple applications using the same device driver essentially you "route" audio from one application to another using an additional asio device. A number of daws have great virtual asio optoins which you can use to do this without an additional physical device. The example we provide is with reaper with "rearoute" which is their virtual asio device. So for example if you wanted to preprocess the audio before it hits obs, or you'd like to separately record audio with another application you could use reaper, send audio through rearoute, and then use this plugin to capture the rearoute virtual asio device.
